services/app-api/handlers/reports/update.ts

```diff
@@ -14,132 +12,185 @@ import { StatusCodes, UserRoles } from "../../utils/types/types";
-                Key: `${buckets.FIELD_DATA}/${state}/${fieldDataId}.json`,
-                Body: JSON.stringify(fieldData),
-                ContentType: "application/json",
-              };
-              await s3Lib.put(fieldDataParams);
-
-              // validate report metadata
-              const validatedMetadata = await validateData(
-                metadataValidationSchema,
-                { ...unvalidatedMetadata }
-              );
-              // if metadata passes validation,
-              if (validatedMetadata) {
-                //Delete raw data prior to updating
-                delete currentReport.fieldData;
-                delete currentReport.formTemplate;
-
-                // update record in report metadata table
-                const reportMetadataParams = {
-                  TableName: process.env.MCPAR_REPORT_TABLE_NAME!,
-                  Item: {
-                    ...currentReport,
-                    ...validatedMetadata,
-                    lastAltered: Date.now(),
-                  },
-                };
-                await dynamoDb.put(reportMetadataParams);
-
-                // set response status and body
-                status = StatusCodes.SUCCESS;
-                body = {
-                  ...reportMetadataParams.Item,
-                  fieldData,
-                  formTemplate,
-                };
-              } else {
-                status = StatusCodes.BAD_REQUEST;
-                body = error.INVALID_DATA;
-              }
-            } else {
-              status = StatusCodes.BAD_REQUEST;
-              body = error.INVALID_DATA;
-            }
-          } else {
-            status = StatusCodes.BAD_REQUEST;
-            body = error.MISSING_DATA;
-          }
-        } else {
-          status = StatusCodes.UNAUTHORIZED;
-          body = error.UNAUTHORIZED;
-        }
-      } else {
-        status = StatusCodes.BAD_REQUEST;
-        body = error.MISSING_DATA;
-      }
-    } else {
-      status = StatusCodes.NOT_FOUND;
-      body = error.NO_MATCHING_RECORD;
-    }
+    return {
+      status: StatusCodes.BAD_REQUEST,
+      body: error.NO_KEY,
+    };
   }
-  return { status, body };
+
+  // If request body is missing, return a 400 error.
+  if (!event?.body) {
+    return {
+      status: StatusCodes.BAD_REQUEST,
+      body: error.MISSING_DATA,
+    };
+  }
+
+  // Ensure user has correct permissions to update a report.
+  if (!hasPermissions(event, [UserRoles.STATE_USER, UserRoles.STATE_REP])) {
+    return {
+      status: StatusCodes.UNAUTHORIZED,
+      body: error.UNAUTHORIZED,
+    };
+  }
+
+  /*
+   * TODO: Is this code necessary given the discrete archive route?
+   * if (hasPermissions(event, [UserRoles.ADMIN])) {
+   *  const { statusCode: archiveStatus, body: archiveBody } =
+   *    await archiveReport(event, context);
+   *  return {
+   *    status: archiveStatus,
+   *    body: JSON.parse(archiveBody),
+   *  };
+   * }
+   */
```

Comment:
Not as such, but we will to check if the user is admin in the archive handler and throw unauthorized if not, similar to how we check permissions at the beginning of the others. 👍

Reply:
Removed this comment block. Thanks @braxex